Formula/perl.rb

```diff
@@ -25,6 +26,11 @@ class Perl < Formula
   depends_on "gdbm"
 
   uses_from_macos "expat"
+  uses_from_macos "libxcrypt"
+
+  on_linux do
+    depends_on "libnsl"
+  end
```

Comment:
There is no linkage with `libnsl`, so this dependency doesn't seem to do anything. The previous run without this also did not register linkage with any `libnsl`, as far as I can tell.

If linkage with a system `libnsl` is not desired, then `brew linkage` needs to be adjusted to make sure it errors.

Reply:
Oh weird, I guess it depends on the system then.  On the OS I was testing this on it tried to find `libnsl` but I guess it doesn't do so depending on the host.  

I think we actually need to adjust `brew linkage` to reject both `libnsl.so.1` and `libcrypt.so.1` as those are both still in the allowlist right now.

I guess better question now is, how do we stop `perl` from trying to link to `libnsl` if we don't want to add it as a dependency?  It may be the case that this only happens when building from source on older systems but I'm not sure.

Reply:
I am okay with dropping `libnsl` for now so we can merge the `libxcrypt` dependency as that is more critical.  Does that sound okay?

Reply:
> I guess better question now is, how do we stop `perl` from trying to link to `libnsl` if we don't want to add it as a dependency?

It might be a configure argument. If there isn't, I'd leave it alone. It's not really that important to stop opportunistic linkage, especially if it doesn't result in a build failure of some kind.

Reply:
Actually it does result in a build failure which was why I wanted to fix this.  Sorry I should have said that earlier!

Reply:
Regrettably the `Configure` script hardcodes the list of libraries it searches for and there doesn't seem to be a way to change this, so I just removed it from the list with `inreplace`.

Reply:
Nvm, I dug up some old stuff from Linuxbrew on how to override `libswanted`.  The syntax is a little wacky but it does the job and avoids any use of `inreplace`.